cilium-health/launch/endpoint.go

```diff
@@ -0,0 +1,213 @@
+// Copyright 2017 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package launch
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/cilium/cilium/api/v1/models"
+	"github.com/cilium/cilium/common/addressing"
+	"github.com/cilium/cilium/common/plugins"
+	"github.com/cilium/cilium/pkg/endpoint"
+	"github.com/cilium/cilium/pkg/endpointmanager"
+	"github.com/cilium/cilium/pkg/k8s"
+	"github.com/cilium/cilium/pkg/labels"
+	"github.com/cilium/cilium/pkg/logging/logfields"
+	"github.com/cilium/cilium/pkg/node"
+	"github.com/cilium/cilium/pkg/option"
+
+	"github.com/vishvananda/netlink"
+)
+
+func logFromCommand(cmd *exec.Cmd, netns string) error {
+	scopedLog := log.WithField("netns", netns)
+	stdout, err := cmd.StdoutPipe()
+	if err != nil {
+		return err
+	}
+	go func() {
+		in := bufio.NewScanner(stdout)
+		for in.Scan() {
+			scopedLog.Debug("%s", in.Text())
```

Comment:
```
scopedLog.Debugf("%s", in.Text())
```
or:
```
scopedLog.Debug(in.Text())
```

Reply:
This trips me up all the time. Having both `Debug()` and `Debugf()` is just asking for people to mis-use the two functions --- particularly since `Debug()` accepts multiple arguments (!). I'd rather consistently use `Debugf()` and not have to care which variation I'm using. (... I'll fix this up)